src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java

```diff
@@ -729,11 +768,14 @@ private boolean isNotRelevantSituation(DetailAST ast, int currentType) {
         final boolean emptyBlockOrType =
                 isEmptyBlock(ast, parentType)
                     || allowEmptyTypes && isEmptyType(ast);
+        final boolean isArrayInitializerAndShouldBeChecked =
```

Comment:
This looks like a cheap way to fool [LogicConditionNeedOptimization](http://sevntu-checkstyle.github.io/sevntu.checkstyle/apidocs/com/github/sevntu/checkstyle/checks/coding/LogicConditionNeedOptimizationCheck.html) check.

For this particular case, it's not important, because the `isArrayInitialization` method is very small and will definitely be inlined by the compiler.

However, it will be more readable if the `enforceWhitespaceAroundArrayInit` field is checked in the `isArrayInitialization` method. The semantics of the method will be changed, so please give it a new good name and description.